examples/all-clusters-app/all-clusters-common/src/static-supported-temperature-levels.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/config.h>
+#include <static-supported-temperature-levels.h>
+
+using namespace std;
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::TemperatureControl;
+using chip::Protocols::InteractionModel::Status;
+
+// TODO: Configure your options for each endpoint
+chip::CharSpan StaticSupportedTemperatureLevels::temperatureLevelOptions[] = { chip::CharSpan("Cold", 4), chip::CharSpan("Warm", 4),
+                                                                               chip::CharSpan("Hot", 3) };
+
+const StaticSupportedTemperatureLevels::EndpointPair
+    StaticSupportedTemperatureLevels::supportedOptionsByEndpoints[EMBER_AF_TEMPERATURE_CONTROL_CLUSTER_SERVER_ENDPOINT_COUNT] = {
+        EndpointPair(1, StaticSupportedTemperatureLevels::temperatureLevelOptions, 3) // Options for Endpoint 1
+    };
+
+int StaticSupportedTemperatureLevels::Size()
+{
+    for (auto & endpointPair : StaticSupportedTemperatureLevels::supportedOptionsByEndpoints)
+    {
+        if (endpointPair.mEndpointId == mEndpoint)
+        {
+            return endpointPair.mSize;
+        }
+    }
+    return 0;
+}
+
+bool StaticSupportedTemperatureLevels::Next(chip::MutableCharSpan & item)
+{
+    for (auto & endpointPair : StaticSupportedTemperatureLevels::supportedOptionsByEndpoints)
+    {
+        if (endpointPair.mEndpointId == mEndpoint)
+        {
+            if (endpointPair.mSize > mIndex)
+            {
+                CHIP_ERROR err = chip::CopyCharSpanToMutableCharSpan(endpointPair.mTemperatureLevels[mIndex], item);
+                if (err != CHIP_NO_ERROR)
```

Comment:
I would think this would return false at least.  Better yet, actually return the error.

Reply:
Returned error